src/device/device.c

```diff
@@ -44,6 +44,46 @@ static void write_open_bus(void* opaque, uint32_t address, uint32_t value, uint3
 {
 }
 
+static unsigned int dd_rom_dma_read(void* opaque, const uint8_t* dram, uint32_t dram_addr, uint32_t cart_addr, uint32_t length)
+{
+    return /* length / 8 */0x1000;
+}
+
+static unsigned int dd_rom_dma_write(void* opaque, uint8_t* dram, uint32_t dram_addr, uint32_t cart_addr, uint32_t length)
+{
+    return /* length / 8 */0x1000;
+}
+
+
+static void get_pi_dma_handler(struct device* dev, uint32_t address, void** opaque, const struct pi_dma_handler** handler)
+{
+#define RW(o, x) \
+    do { \
+    static const struct pi_dma_handler h = { x ## _dma_read, x ## _dma_write }; \
+    *opaque = (o); \
+    *handler = &h; \
+    } while(0)
+
+    if (address >= MM_CART_ROM) {
+        if (address >= MM_CART_DOM3) {
+            /* 0x1fd00000 - 0x7fffffff : dom3 addr2, cart ??? */
+            RW(NULL, cart_dom3);
+        }
+        else {
+            /* 0x10000000 - 0x1fbfffff : dom1 addr2, cart rom */
```

Comment:
Should this comment say 0x1fcffff.....? It says dom3 starts at 0x1fd...

Reply:
0x1fc00000 is allocated to pif rom, pif RAM. Cartridge domain 3 does indeed start at 0x1fd00000, but I don't know what is mapped there and how it is accessed.

Reply:
Ah ok makes sense